### PR TITLE
feat: add plugins to manual relations

### PR DESCRIPTION
**Pull request by jackd248**
Solves #25

### CODE REVIEW

**Comment by jackd248** [comment]
You're right, but I didn't get the idea behind this. Where can I find the feature to link manual content elements with relations?

**Comment by maikschneider** [comment]
You can link single `mtext` elements: Just like pages there is a tab for relation setting. I think this is useful when a manual page contains plenty of elements and the related one is at the very bottom.

**Comment by jackd248** [comment]
Alright, I will complete the missing stuff.

File: Classes/UserFunctions/SelectItemsProcFunc.php [hunk 1]
```diff
@@ -100,4 +101,20 @@ public static function getIconForTableAndType(string $table, string|int $type):
 
         return $GLOBALS['TCA'][$table]['ctrl']['iconfile'] ?? '';
     }
+
+    public static function getPlugins(): array
```
Comment (maikschneider) [inline Classes/UserFunctions/SelectItemsProcFunc.php:105]: All class members (not invoke) can be protected. Could you please adjust?

File: Classes/EventListener/ModifyButtonBarEventListener.php [hunk 2]
```diff
@@ -78,34 +82,58 @@ public function getManualPages(string $uri, ServerRequestInterface $request, int
...
+        return $this->getRecords(
+            sprintf(
+                'select p.uid, p.title from tt_content r, pages p where r.uid=%s and FIND_IN_SET(concat("tt_content:list:", r.list_type), (p.tx_ximatypo3manual_relations)) and p.deleted=0 and p.hidden=0 and r.CType="list" and r.list_type!=""',
+                $recordUid
+            ),
+            'pages'
+        );
+    }
+
+    protected function getRecords(string $sql, string $table): array
```
Comment (maikschneider) [inline Classes/EventListener/ModifyButtonBarEventListener.php:133]: The name is misleading. Could you please change it into something like executeSql, fetchRecords,..

**Review by maikschneider — CHANGES_REQUESTED** [review]
Nice. Just some small adjustments

**Comment by maikschneider** [comment]
@jackd248 appreciate your input, thank you very much!

**Comment by jackd248** [comment]
Enjoy your holidays!

**Pull request merged**